networkx/algorithms/isomorphism/tree_isomorph.py

```diff
@@ -0,0 +1,48 @@
+"""
+Tree isomorphism functions.
+"""
+import networkx as nx
+from networkx.exception import NetworkXError
+
+__author__ = """\n""".join(['Balazs Kossovics (kossovics@gmail.com)',
+    'Jeffrey Finkelstein (jeffrey.finkelstein@gmail.com)'])
+
+__all__ = ['rooted_tree_is_isomorphic',
+           'tree_is_isomorphic']
+
+# based on http://crypto.cs.mcgill.ca/~crepeau/CS250/2004/HW5+.pdf
+#   and jfinkels' comment
+# usage:
+# T1 = nx.random_powerlaw_tree(10, tries=1000)
+# T2 = nx.random_powerlaw_tree(10, tries=1000)
+# print tree_is_isomorphic(T1, T2)
+
+def canonical_form(T, root, _from=None):
+    try:
+        children = filter(lambda node: node != _from, T[root])
+        if len(children) == 0:
+            return ()
+        return tuple(sorted(canonical_form(T, v, root) for v in children))
+    except KeyError:
+        raise NetworkXError("Graph {} contains no node {}".format(T, root))
+
+def rooted_tree_is_isomorphic(T1, T2, r1, r2):
+    if not nx.is_tree(T1):
+        raise NetworkXError("Graph {} is not a tree.".format(T1))
+
+    if not nx.is_tree(T2):
+        raise NetworkXError("Graph {} is not a tree.".format(T2))
+
+    return canonical_form(T1, r1) == canonical_form(T2, r2)
+
+
+def tree_is_isomorphic(T1, T2):
```

Comment:
I think it would be better to have these as a single function, `is_isomorphic()` in the module `isomorphism/tree.py`, with keyword arguments for the (optional) roots:

    def is_isomorphic(T1, T2, root1=None, root2=None):
        if root1 is not None and root2 is not None:
            return canonical_form(...) == ...
        if root1 is None and root2 is None:
            return any(is_isomorphic(T1, T2, r1, r2) for r1 in center(T1) for r2 in center(T2)))
        raise Exception